pkg/apis/core/types.go

```diff
@@ -3348,6 +3348,23 @@ type NodeSystemInfo struct {
 	Architecture string
 }
 
+// NodeConfigStatus describes the status of the config assigned by Node.Spec.ConfigSource
+type NodeConfigStatus struct {
+	// Assigned is the node's ack of the config source assigned via Node.Spec.ConfigSource,
+	// and is a direct reflection of that source.
+	// +optional
+	Assigned *NodeConfigSource
```

Comment:
can you clarify the circumstances when Assigned != Active, and when Active != LastKnownGood?

Reply:
for example, it would be helpful to know things like this (these are just my imaginings, review for accuracy and for the implications of those states):

1. when the node has observed an updated config, has validated it, but has not yet made it active, Assigned will reflect the new config and Active will reflect the previous, still-running config. The node will not attempt to use the referenced config again until it is updated.
2. when the node has observed an updated config, cannot validate it, and will not make it active, Assigned will reflect the new config, Active will reflect the previous, still-running config, and Error will contain the validation error. The node will not attempt to use the referenced config again until it is updated.
3. when the node started with an updated config, encountered errors, and reverted to a LastKnownGood config, Assigned will reflect the new config, Active will contain the LastKnownGood config, and Error will contain a description of the error encountered. The node will not attempt to use the referenced config again until it is updated.
4. when the node has started with an updated config, but has not yet run long enough to verify the config is good, LastKnownGood will reference the previous known good config (if any exists).

Reply:
That's mostly correct. 

With (2), `Active` remains unchanged if the error happens prior to the Kubelet restarting to use the new config (e.g. some error prevented downloading or checkpointing), but will change to `LastKnownGood` if the Kubelet progresses to a restart and then encounters an error loading, parsing, or validating the unpacked payload (in the future, I'd like to optimize the flow by trying a load/parse/validate prior to restart, but this is how things work today).

Reply:
Updated, let me know if you see other edges I should clarify.